src/NzbDrone.Core/Download/ProcessDownloadDecisions.cs

```diff
@@ -36,52 +38,91 @@ public ProcessedDecisions ProcessDecisions(List<DownloadDecision> decisions)
             var prioritizedDecisions = _prioritizeDownloadDecision.PrioritizeDecisions(qualifiedReports);
             var grabbed = new List<DownloadDecision>();
             var pending = new List<DownloadDecision>();
+            var storedUsenet = new List<DownloadDecision>();
+            var fallbackUsenet = new List<DownloadDecision>();
+            var storedTorrent = new List<DownloadDecision>();
+            var fallbackTorrent = new List<DownloadDecision>();
 
             foreach (var report in prioritizedDecisions)
             {
                 var remoteEpisode = report.RemoteEpisode;
+                var downloadProtocol = report.RemoteEpisode.Release.DownloadProtocol;
 
-                var episodeIds = remoteEpisode.Episodes.Select(e => e.Id).ToList();
-
-                //Skip if already grabbed
-                if (grabbed.SelectMany(r => r.RemoteEpisode.Episodes)
-                                .Select(e => e.Id)
-                                .ToList()
-                                .Intersect(episodeIds)
-                                .Any())
+                // Skip if already grabbed
+                if (IsEpisodeProcessed(grabbed, report))
                 {
                     continue;
                 }
 
                 if (report.TemporarilyRejected)
                 {
-                    _pendingReleaseService.Add(report);
+                    _pendingReleaseService.Add(report, PendingReleaseReason.Delay);
                     pending.Add(report);
                     continue;
                 }
 
-                if (pending.SelectMany(r => r.RemoteEpisode.Episodes)
-                        .Select(e => e.Id)
-                        .ToList()
-                        .Intersect(episodeIds)
-                        .Any())
+                if (IsEpisodeProcessed(pending, report))
```

Comment:
It already was this way, but I wonder, wont this actually cause undesired behavior?
If one download is Delayed, for example because the user specified a Minimum Age for Usenet, it would cause the Torrent downloads to be delayed as well.